src/test/Microsoft.DotNet.CoreSetup.Packaging.Tests/WindowsDesktopTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.CoreSetup.Test;
+using System.ComponentModel.Design;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Packaging.Tests
+{
+    public class WindowsDesktopTests
+    {
+        private readonly RepoDirectoriesProvider dirs = new RepoDirectoriesProvider();
+
+        [Fact]
+        public void WindowsDesktopTargetingPackIsValid()
+        {
+            using (var tester = NuGetArtifactTester.OpenOrNull(
+                dirs,
+                "Microsoft.WindowsDesktop.App.Ref"))
+            {
+                if (ShouldExist)
```

Comment:
IMO the readability of this could be improved.  I suspect most new readers of this code will be forced to look at the implementation to figure out what this means.

Reply:
Renamed to `CurrentRidShouldCreateNupkg` (pointless to be generic about it until there's a reason to be).